client/my-sites/checkout/src/components/wp-order-review-line-items.tsx

```diff
@@ -93,8 +94,18 @@ export function WPOrderReviewLineItems( {
 	const creditsLineItem = getCreditsLineItemFromCart( responseCart );
 	const couponLineItem = getCouponLineItemFromCart( responseCart );
 	const isOnboardingAffiliateFlow = useSelector( getIsOnboardingAffiliateFlow );
-	if ( isOnboardingAffiliateFlow && couponLineItem ) {
-		couponLineItem.label = getAffiliateCouponLabel();
+	const productSlugs = responseCart.products?.map( ( product ) => product.product_slug );
+	const [ , experimentAssignment ] = useExperiment( 'calypso_checkout_hide_coupon_box', {
+		isEligible: ! productSlugs.some( ( slug ) => 'wp_difm_lite' === slug ),
+	} );
+
+	if ( couponLineItem ) {
+		couponLineItem.label = isOnboardingAffiliateFlow
+			? getAffiliateCouponLabel()
+			: getCouponLabel( couponLineItem.label, experimentAssignment?.variationName || null );
```

Comment:
wouldn't just passing  `experimentAssignment?.variationName` worked here ?

Reply:
The issue was that` experimentAssignment? variationName` can also return undefined, which generated a Typescript error. I adjusted the function to accept `null` | `undefined`, so the code is a bit cleaner.